controllers/code.go

```diff
@@ -1,24 +1,43 @@
 package controllers
 
 import (
+	"errors"
+	"io/ioutil"
 	"net/http"
 
+	"github.com/coduno/api/util"
+	"github.com/gorilla/mux"
+
+	"google.golang.org/cloud/storage"
+
 	"golang.org/x/net/context"
 )
 
-// DownloadTemplate serves a static file to a client.
+// Template serves the contents of a static file to a client.
 // TODO(flowlo, victorbalan): Decide where the templates will be stored.
-func Template(c context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
+func Template(ctx context.Context, w http.ResponseWriter, r *http.Request) (int, error) {
 	if r.Method != "GET" {
 		return http.StatusMethodNotAllowed, nil
 	}
 
-	// TODO(victorbalan): Serve correct template using passenger and result key.
+	templateName := mux.Vars(r)["name"]
+	language := mux.Vars(r)["language"]
+	fileName, ok := util.FileNames[language]
```

Comment:
Name and language is not specific enough. We'll have to provide a template per-task.